From dbb4b2f02d0c54e33c836785eb8b4974720801f3 Mon Sep 17 00:00:00 2001
From: Andrew Mickelson <andrew.mickelson@gmail.com>
Date: Thu, 23 Nov 2017 18:11:37 -0800
Subject: [PATCH] Issue #385 - fix recent compile fix for older versions of
 avcodec

Old versions of ffmpeg/libav defined FF_INPUT_BUFFER_PADDING_SIZE.
That got switched to AV_INPUT_BUFFER_PADDING_SIZE.
The most recent versions seem to have removed the FF_ version.

This commit should address https://github.com/mickelson/attract/commit/4b93a5276fae4b6cdd00d222d9d5b4ba5e02af07
breaking the build on old ffmpeg/libav versions
---
 src/media.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/media.cpp b/src/media.cpp
index 6f7f131..69ae888 100644
--- a/src/media.cpp
+++ b/src/media.cpp
@@ -73,6 +73,14 @@ extern "C"
 #include <queue>
 #include <iostream>
 
+//
+// As of Nov, 2017 RetroPie's default version of avcodec is old enough
+// that it doesn't define AV_INPUT_PADDING_SIZE
+//
+#ifndef AV_INPUT_BUFFER_PADDING_SIZE
+#define AV_INPUT_BUFFER_PADDING_SIZE FF_INPUT_BUFFER_PADDING_SIZE
+#endif
+
 void print_ffmpeg_version_info()
 {
 	FeLog() << "avcodec " << LIBAVCODEC_VERSION_MAJOR
From 4b93a5276fae4b6cdd00d222d9d5b4ba5e02af07 Mon Sep 17 00:00:00 2001
From: Andrew Mickelson <andrew.mickelson@gmail.com>
Date: Mon, 20 Nov 2017 18:51:43 -0800
Subject: [PATCH] Issue #385 - fix compile error on Jessie

---
 src/media.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/media.cpp b/src/media.cpp
index b762e57..6f7f131 100644
--- a/src/media.cpp
+++ b/src/media.cpp
@@ -970,11 +970,11 @@ bool FeMedia::openFromArchive( const std::string &archive,
 
 	size_t avio_ctx_buffer_size = 4096;
 	uint8_t *avio_ctx_buffer = (uint8_t *)av_malloc( avio_ctx_buffer_size
-			+ FF_INPUT_BUFFER_PADDING_SIZE );
+			+ AV_INPUT_BUFFER_PADDING_SIZE );
 
 	memset( avio_ctx_buffer + avio_ctx_buffer_size,
 		0,
-		FF_INPUT_BUFFER_PADDING_SIZE );
+		AV_INPUT_BUFFER_PADDING_SIZE );
 
 	m_imp->m_io_ctx = avio_alloc_context( avio_ctx_buffer,
 		avio_ctx_buffer_size, 0, z, &fe_zip_read,
@@ -1030,7 +1030,7 @@ bool FeMedia::internal_open( sf::Texture *outt )
 				//
 				m_audio->buffer = (sf::Int16 *)av_malloc(
 					MAX_AUDIO_FRAME_SIZE
-					+ FF_INPUT_BUFFER_PADDING_SIZE
+					+ AV_INPUT_BUFFER_PADDING_SIZE
 					+ codec_ctx->sample_rate );
 
 				sf::SoundStream::initialize(
